src/services/oss/backend.rs

```diff
@@ -452,13 +490,13 @@ impl Backend {
         Ok(req)
     }
 
-    fn oss_head_object_request(&self, path: &str) -> Result<Request<AsyncBody>> {
+    fn oss_head_object_request(&self, path: &str, use_presign: bool) -> Result<Request<AsyncBody>> {
         let p = build_abs_path(&self.root, path);
-
-        let url = format!("{}/{}", self.endpoint, percent_encode_path(&p));
+        let (endpoint, host) = self.choose_endpoint_and_host(use_presign);
+        let url = format!("{}/{}", endpoint, percent_encode_path(&p));
 
         let mut req = Request::head(&url);
-        req = req.header(HOST, &self.host);
+        req = req.header(HOST, host);
```

Comment:
We don't need to set host for requests, we can remove all related code.

Reply:
thank you for your advice, these `req.header(Host, self.host)` have been cleaned up.